rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -461,26 +470,31 @@ def _tag_id_index_mapping_for(
 
         return tag_id_dict
 
-    def _uses_sequence_features_for_text(self) -> bool:
+    def _uses_sequence_features_for_input_text(self) -> bool:
         """Whether we need sequence features for the `TEXT` attribute.
 
         Note that, for prediction of some label attribute, the `DIETClassifier`
         can make use of sequence features even if the number of
         transformer layers is 0 because for that it creates a BOW representation
         by simply summing over sequence+sentence features.
         """
-        return (
-            self.component_config[NUM_TRANSFORMER_LAYERS] > 0
-            or self.component_config[ENTITY_RECOGNITION]
-            or self.label_attribute is not None
-        )
+        return True
+
+    def _needs_sentence_features_for_input_text(self) -> bool:
```

Comment:
because _uses is optional (i.e. we drop sequence features which we always assume to be there if the architecture won't use them - i.e. results won't differ if you changed those) and _needs means they need this to function properly and we have a problem if they're missing